cfblog/utils.py

```diff
@@ -0,0 +1,283 @@
+__author__ = 'vinay'
+import datetime
+from functools import wraps
+import os
+import re
+
+from bs4 import BeautifulSoup, Tag
+from mistune import markdown
+
+from django.contrib.auth.models import AnonymousUser
+from django.core.cache import cache
+from django.core.exceptions import PermissionDenied
+from django.core.urlresolvers import resolve
+from django.http.request import HttpRequest
+from django.template import TemplateSyntaxError
+from django.utils.decorators import available_attrs
+from django.utils.text import slugify
+
+from .exceptions import CmsInvalidTemplateException, CmsInvalidContent
+from .settings import CMS_AUTH_TEST_FUNC, PAGE_CACHE_TIMEOUT
+from .validators import validate_and_get_template, ValidationError
+
+dum_request = HttpRequest()
+dum_request.user = AnonymousUser()
+
+CMS_ATTRIBUTES = ['data-cms-attr', 'data-cms-content', 'data-cms-include']
+
+attr_re = re.compile(r'^(?:(?:[a-z-]+:[a-z_0-9]+)+(?:\|(?:[a-z-]+:[a-z_0-9]+))*)*$', re.IGNORECASE)
+content_re = re.compile(r'^(?:md:)?([a-z_0-9])+$', re.IGNORECASE)
+include_html_re = re.compile(r'^(?:(?:[a-z_0-9])+:)?.+$', re.IGNORECASE)
+namespace_re = re.compile(r'^(?:[a-z_0-9])+$', re.IGNORECASE)
+
+NAMESPACE_DELIMITER = '-'
```

Comment:
Can we use some other namespace delimiter? - is never a namespace delimiter and will be part of many template names. ":" or "|" is better.